### PR TITLE
🐛 [RUM-87] AddError should support Error instances coming from other JS contexts

### PR DESCRIPTION
**Pull request by RomanGaignault**
## Motivation

using DD_RUM.addError(error) but the error comes from a different context than the document (ex: created insde an iframe or a window.open), the SDK doesn not consider it as an Error instance and serializes it as JSON.

## Changes

Use class object to determine if it's an error or not.

## Testing

- [ ] Local
- [ ] Staging
- [ ] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/core/src/domain/console/consoleObservable.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ function buildConsoleLog(params: unknown[], api: ConsoleApiName, handlingStack:
   let error: RawError | undefined
 
   if (api === ConsoleApiName.error) {
-    const firstErrorParam = find(params, (param: unknown): param is Error => param instanceof Error)
+    const firstErrorParam = find(params, (param: unknown): param is Error => isError(param))
```
Comment (BenoitZugmeyer) [inline packages/core/src/domain/console/consoleObservable.ts:77]: **💬 suggestion:** ‏
```suggestion
    const firstErrorParam = find(params, isError)
```

File: packages/core/src/domain/error/error.ts [hunk 2]
```diff
@@ -80,19 +80,21 @@ function hasUsableStack(isErrorInstance: boolean, stackTrace?: StackTrace): stac
...
+  return isError(originalError) && 'dd_fingerprint' in originalError ? String(originalError.dd_fingerprint) : undefined
 }
 
 export function getFileFromStackTraceString(stack: string) {
   return /@ (.+)/.exec(stack)?.[1]
 }
 
+export function isError(error: unknown): error is Error {
+  return Object.prototype.toString.call(error) === '[object Error]'
+}
```
Comment (BenoitZugmeyer) [inline packages/core/src/domain/error/error.ts:92]: **💬 suggestion:** ‏Add a test case for this. Something like:

```js
const iframe = document.createElement('iframe');
document.body.appendChild(iframe);
registerCleanupTask(() => document.body.removeChild(iframe))
expect(isError(new iframe.contentWindow.Error()).toBe(true)
```

**Review by BenoitZugmeyer — APPROVED** [review]

**Comment by amortemousque** [comment]
Should we do the same everytime we check `instanceof Error``
Like in the logger: https://github.com/DataDog/browser-sdk/blob/f90ede411e67ec3cdba264e9f1a8cdef5321088f/packages/logs/src/domain/logger.ts#L67
in the telemetry: https://github.com/DataDog/browser-sdk/blob/f90ede411e67ec3cdba264e9f1a8cdef5321088f/packages/core/src/domain/telemetry/telemetry.ts#L205

File: packages/logs/src/domain/logger.ts [hunk 3]
```diff
@@ -64,7 +65,7 @@ export class Logger {
 
     if (error !== undefined && error !== null) {
       const rawError = computeRawError({
-        stackTrace: error instanceof Error ? computeStackTrace(error) : undefined,
+        stackTrace: isError(error) ? computeStackTrace(error) : undefined,
```
Comment (amortemousque) [inline packages/logs/src/domain/logger.ts:68]: **💬 suggestion:** Could be nice to have a test about it.‏

**Review by BenoitZugmeyer — APPROVED** [review]

**Pull request merged**